src/App.hs

```diff
@@ -64,3 +70,17 @@ runMatterhorn opts config = do
       Just h -> hClose h
 
     return finalSt
+
+-- | Cleanup resources and save data for restoring on program restart.
+closeMatterhorn :: ChatState -> IO ()
+closeMatterhorn finalSt = do
+  mmCloseSession (finalSt^.csResources.crSession)
```

Comment:
Please make sure this can't raise exceptions and prevent the rest of the function from running.